### PR TITLE
chore: i18n cleanup script

### PR DESCRIPTION
**Pull request by randal923**
## What does this PR do?

- Fixes #15623
- /claim #15623
- 
## Mandatory Tasks (DO NOT REMOVE)

- [ x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [ x] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com). If N/A, write N/A here and check the checkbox.
- [ x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

Make sure all the correct unused keys are being removed. Also, make sure to add any keys that should not be removed to the KEYS_TO_NOT_REMOVE array.

I initially considered two approaches. The first was to use glob to return a string of each component and then match regex patterns for i18n, t(), and other patterns. The second approach involved parsing the Abstract Syntax Tree (AST) and looking for these patterns to remove. But, I noticed that some parts of the code weren't using these patterns at all, so I decided to take a different route. Now, I'm first getting the keys from the common files and then searching for them in the string of the component that glob provides and removing them if they don't exist.

### CODE REVIEW

**Comment by randal923** [comment]
Looks like me merging the main to fix a conflict also fixed the formatting of 3 unrelated files. Should I remove them?

**Review by Amit91848 — CHANGES_REQUESTED** [review]
Great stuff @randal923 Looks promising. I haven't looked at the code yet but seems to be working well. 

You would also have to sign the CLA, without it we cannot merge this.

Small issues I found:
- Unit test that you added are failing, please look into it.
- We have a few strings that are being stored directly in the database, we would like to keep those.
  Screenshot 2024-08-01 133921
- "ADD_NEW_STRINGS_ABOVE_THIS_LINE_TO_PREVENT_MERGE_CONFLICTS": This line is getting removed from all translations. Please ignore this, we want it to be there in the all files. 
- Getting an error in /lv/common.json
Screenshot 2024-08-01 134603

File: apps/web/scripts/i18n-cleanup/main.ts
Comment (Amit91848) [inline apps/web/scripts/i18n-cleanup/main.ts:19]: Might also want to include sql files here as we have some keys directly stored in db

File: apps/web/scripts/i18n-cleanup/main.ts
Comment (Amit91848) [inline apps/web/scripts/i18n-cleanup/main.ts:10]:
```suggestion
export const KEYS_TO_NOT_REMOVE: readonly string[] = [
  "ADD_NEW_STRINGS_ABOVE_THIS_LINE_TO_PREVENT_MERGE_CONFLICTS",
] as const;
```

File: apps/web/scripts/i18n-cleanup/utils.ts
Comment (Amit91848) [inline apps/web/scripts/i18n-cleanup/utils.ts:17]: Need to leave a line at the end of the file

File: apps/web/scripts/i18n-cleanup/main.ts
Comment (Amit91848) [reply apps/web/scripts/i18n-cleanup/main.ts:19]:
```suggestion
  const sourceFiles = glob.sync(`${sourceDir}/**/*.{ts,tsx,js,jsx,sql}`, {
```

**Comment by randal923** [comment]
Hi @Amit91848, thanks for the feedback. I will work on the changes.

**Comment by randal923** [comment]
@Amit91848 I think these commits fixes everything you asked for. Let me know if there is anything else I need to change.

File: apps/web/scripts/i18n-cleanup/main.ts
Comment (Amit91848) [inline apps/web/scripts/i18n-cleanup/main.ts:21]:
```suggestion
  const sourceFiles = glob.sync(`${sourceDir}/**/*.{ts,tsx,js,jsx,sql}`, {
```

**Review by Amit91848 — COMMENTED** [review]
Works fine now. Only removes the un used translation keys. 

It is a lot slower than the PR we have, maybe if we parallelize scanning files and use regex instead of  .includes, it could be a lot faster. But wait for a second review from senior team members before making any changes and let them decide.

**Comment by randal923** [comment]
> Works fine now. Only removes the un used translation keys.
> 
> It is a lot slower than the PR we have, maybe if we parallelize scanning files and use regex instead of .includes, it could be a lot faster. But wait for a second review from senior team members before making any changes and let them decide.

Using regex won't clean all the unused string because some of them don't follow any of the i18n library patterns (t(), i18n(), etc...)

**Pull request closed without merging**
